### PR TITLE
Fixed warnings

### PR DESCRIPTION
**Pull request by babrodtk**
This fixes several warnings, and should hopefully have no side effects.

### CODE REVIEW

File: opm/autodiff/BlackoilPropsAdFromDeck.cpp [hunk 1]
```diff
@@ -40,6 +40,18 @@
 #include <opm/parser/eclipse/Deck/Deck.hpp>
 #include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
 
+
+namespace {
+    inline
+    void silenceCompilerWarnings() {
```
Comment (atgeirr) [inline opm/autodiff/BlackoilPropsAdFromDeck.cpp]: I have seen warnings for these, but strangely only when there are compiler errors. Are you seeing warnings for these even for a successful compile?

Also: why is it placed in this file? It should probably be placed in opm-core's Units.hpp if it is required.

File: opm/autodiff/ExtractParallelGridInformationToISTL.hpp [hunk 2]
```diff
@@ -21,9 +21,13 @@
 #define OPM_EXTRACTPARALLELGRIDINFORMATIONTOISTL_HEADER_INCLUDED
 #ifdef HAVE_DUNE_CORNERPOINT
 
+#include <opm/core/utility/platform_dependent/disable_warnings.h>
+
```
Comment (atgeirr) [inline opm/autodiff/ExtractParallelGridInformationToISTL.hpp]: The CpGrid include should be outside the warning fences.

File: opm/autodiff/BlackoilPropsAdFromDeck.cpp (hunk 1 above)
Comment (babrodtk) [inline opm/autodiff/BlackoilPropsAdFromDeck.cpp]: Well made comments. I do not see these for successful builds, so perhaps the proper solution is to remove this function altogether?

File: opm/autodiff/ExtractParallelGridInformationToISTL.hpp (hunk 2 above)
Comment (babrodtk) [inline opm/autodiff/ExtractParallelGridInformationToISTL.hpp]: Thanks for spotting this, that made me miss ~20 warnings

File: opm/autodiff/BlackoilPropsAdFromDeck.cpp (hunk 1 above)
Comment (atgeirr) [inline opm/autodiff/BlackoilPropsAdFromDeck.cpp]: An explanation as to why we see warnings occasionally for un-successful builds would be nice, but I do not have one. Given that successful builds do not produce a warning, I think it is best to remove this.

**Comment by babrodtk** [comment]
@atgeirr I fixed up what you commented.

**Comment by atgeirr** [comment]
I think you maybe were a little too quick: boost.any should still be within the warning fences.

**Comment by babrodtk** [comment]
I received no warnings from boost, so deliberately left it out. If you are experiencing warnings from there (or think its better to have the guards present), I'd be happy to add them again :)

**Comment by atgeirr** [comment]
OK, we'll add them if necessary.

**Pull request merged**